binding/flutter/lib/porcupine.dart

```diff
@@ -92,121 +106,116 @@ class Porcupine {
   /// Thows a `PvError` if not initialized correctly
   ///
   /// returns an instance of the wake word engine
-  static Future<Porcupine> fromKeywordPaths(List<String?>? keywordPaths,
-          {String? modelPath, List<double>? sensitivities}) =>
-      Porcupine._create(
-          keywordPaths: keywordPaths,
-          modelPath: modelPath,
-          sensitivities: sensitivities);
+  static Future<Porcupine> fromKeywordPaths(
+      String accessKey, List<String?>? keywordPaths,
+      {String? modelPath, List<double>? sensitivities}) async {
+    await _init();
+
+    return Porcupine._create(
+        accessKey: accessKey,
+        keywordPaths: keywordPaths,
+        modelPath: modelPath,
+        sensitivities: sensitivities);
+  }
 
   /// private static creator
   static Future<Porcupine> _create(
-      {List<String?>? keywords,
+      {String? accessKey,
       List<String?>? keywordPaths,
       String? modelPath,
       List<double>? sensitivities}) async {
-    if (!_resourcesExtracted) {
-      await _extractPorcupineResources();
-      _resourcesExtracted = true;
+    if (accessKey == null || accessKey == "") {
+      throw PorcupineInvalidArgumentException("'accessKey' must be set.");
     }
 
     if (modelPath == null || modelPath == "") {
       modelPath = _defaultModelPath;
```

Comment:
we can probably let empty model path through and let native fill in default model, right?